service/history/visibilityQueueTaskExecutor.go

```diff
@@ -477,6 +481,31 @@ func (t *visibilityQueueTaskExecutor) isCloseExecutionVisibilityTaskPending(task
 	return !queues.IsTaskAcked(queryTask, visibilityQueueState)
 }
 
+// cleanupExecutionInfo cleans up workflow execution info after visibility close
+// task has been processed and acked by visibility store.
+func (t *visibilityQueueTaskExecutor) cleanupExecutionInfo(
+	ctx context.Context,
+	task *tasks.CloseExecutionVisibilityTask,
+) (retError error) {
+	weContext, release, err := getWorkflowExecutionContextForTask(ctx, t.cache, task)
+	if err != nil {
+		return err
+	}
+	defer func() { release(retError) }()
+
+	mutableState, err := weContext.LoadMutableState(ctx)
+	if err != nil {
+		return err
+	}
+
+	executionInfo := mutableState.GetExecutionInfo()
+	executionInfo.Memo = nil
+	executionInfo.SearchAttributes = nil
+	executionInfo.CloseVisibilityTaskCompleted = true
```

Comment:
Those changes should be persisted by calling SetWorkflowExecution (since workflow is already closed updateWorkflowExecution won't work I think)?